xonsh/environ.py

```diff
@@ -7,62 +7,188 @@
 import builtins
 import platform
 import subprocess
+from datetime import datetime
+from functools import partial
 from warnings import warn
 
 from xonsh.tools import TERM_COLORS
 
-def current_branch(cwd=None):
-    """Gets the branch for a current working directory. Returns None
-    if the cwd is not a repository.  This currently only works for git, 
-    bust should be extended in the future.
-    """
-    branch = None
-    cwd = os.getcwd() if cwd is None else cwd
+class PromptFormatter(dict):
```

Comment:
Why is that?  collections.MutableMapping is an abstract base class.  So if we inherit from that we have to implement all of the Mapping methods ourselves.  If we inherit from dict then we only have to implement the methods that we care about.   The one caveat that I know of being that if you implement ```__getitem__()``` you also have to implement ```get()``` because of how dict's get() method is implemented.  (And if we were overriding ```__setitem__()```, similar issues for the other value setting methods).

Inheriting from dict directly is less code to write and maintain, less objects to create (no need for a separate private dict to actually store the items), and somewhat faster methods (since the methods we don't override are implemented in C).  If you know of some problem that ```MutableMapping()`` solves, I can change it, it just doesn't seem to make sense since really we only need a custom ```__getitem__()```  for this to work with python's str.format() method.